pgmax/fg/graph.py

```diff
@@ -216,15 +260,32 @@ def log_potentials(self) -> np.ndarray:
             A jnp array representing the log of the potential function for each
                 valid configuration
         """
+        if not np.any(
```

Comment:
Instead of doing this kind of `if`, I think again it's better to organize things into dictionaries mapping from factor types to elements, and looping through dictionary elements and return a dictionary.

In this case, one dictionary element would be all enumeration factors, and the original function can be applied without change. The other would be all the OR factors, for which we just return None.